website/docs/r/kinesis_firehose_delivery_stream.html.markdown

```diff
@@ -107,7 +107,7 @@ resource "aws_kinesis_firehose_delivery_stream" "extended_s3_stream" {
     role_arn   = aws_iam_role.firehose_role.arn
     bucket_arn = aws_s3_bucket.bucket.arn
 
-    buffering_size = 64
+    buffer_size = 64
```

Comment:
Based on the [documentation](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/kinesis_firehose_delivery_stream#s3-configuration) and [schema](https://github.com/hashicorp/terraform-provider-aws/blob/dfd52e40ea10872efff5ec5b797ebcdcff85585e/internal/service/firehose/delivery_stream.go#L156), `buffering_size` is the correct argument. While this particular line should not be changed, line 166 incorrectly uses `buffer_size`. Was that the line you intended to edit?

```suggestion
    buffering_size = 64
```

Reply:
Hi @justinretzolk, thanks for taking a look but this is not correct. The block `extended_s3_configuration` only accepts `buffer_size` but not `buffering_size`. Line 166 shows `buffer_size` to me so this is correct. Here is a screenshot. 

<img width="665" alt="image" src="https://github.com/hashicorp/terraform-provider-aws/assets/124513/7ba0ccd2-90e2-4f40-b21c-f49190d69b2e">

It's funny that you are linking to the docs I am trying to fix. Also the schema for [this particular block is here](https://github.com/hashicorp/terraform-provider-aws/blob/dfd52e40ea10872efff5ec5b797ebcdcff85585e/internal/service/firehose/delivery_stream.go#L1025) I believe. It still says `buffering_size` so there might be another problem somewhere or I am not getting something.

```
Terraform v1.4.0
on darwin_arm64
+ provider registry.terraform.io/hashicorp/aws v4.58.0
```

Reply:
just tried it with tf `1.5.2` and still the same behavior

Reply:
Hey @benben 👋 It looks like this is a result of your use of AWS Provider version `4.58.0`. In version `5.0.0`, the `buffer_size` argument [was renamed to `buffering_size`](https://github.com/hashicorp/terraform-provider-aws/blob/5ce64c172eff331b0de12e9e86b8773b75eb30a1/CHANGELOG.md?plain=1#L345). Looks like line 166 was missed when this change occurred, which is why it still reads as `buffer_size`.